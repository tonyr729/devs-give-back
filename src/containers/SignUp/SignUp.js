import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import DatabaseHelper from '../../helpers/DatabaseHelper';
import { handleSignup } from '../../actions/actions';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      repo: '',
      contact: ''
    };
    this.database = new DatabaseHelper();
  }

  handleInputChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmit = () => {
    const {repo, contact} = this.state;
    const { name, id } = this.props.dev;
    const { status, problemID} = this.props.signup;
    this.database.writeContributerToDatabase(repo, contact, name, id, problemID);
    this.database.writeDevProjectToDatabase(id, problemID);
    this.props.handleSignup(!status);
  }

  render() {
    const status = this.props.signup.status;
    return (
      <div className="overlay">
        {status || <Redirect to='/dev-profile'/>}
        <div className="signup-form">
          <p className="signup-title">Awesome!</p>
          <p className="signup-info">
            Please create a GitHub Repo for this project 
            and prefered contact info for the client and 
            other developers regarding this project. 
          </p>
          <input type="text" 
            name="repo" 
            placeholder="https://github.com/:user/:repo" 
            className="repo-input" 
            value={this.state.repo} 
            onChange={this.handleInputChange}
            autoFocus/>
          <input type="text" 
            name="contact" 
            placeholder="Contact Info" 
            className="contact-info" 
            value={this.state.contact}
            onChange={this.handleInputChange}/>
          <button onClick={this.handleSubmit} className="submit-signup">Submit!</button>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  dev: state.dev,
  signup: state.signup
});

export const mapDispatchToProps = (dispatch) => ({
  handleSignup: (status) => dispatch(handleSignup(status))
});



export default connect(mapStateToProps, mapDispatchToProps)(SignUp);