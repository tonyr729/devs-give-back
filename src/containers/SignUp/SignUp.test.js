import { SignUp} from './SignUp';
import React from 'react';
import { shallow } from 'enzyme';
jest.mock('../../helpers/DatabaseHelper.js', () => {
  return jest.fn().mockImplementation(() => {
    return {
      writeContributerToDatabase: jest.fn().mockImplementation(() => Promise.resolve({category: 'category'})),
      writeDevProjectToDatabase: jest.fn().mockImplementation(() => Promise.resolve({problem: 'problem'}))
    };
  });
});

describe('SignUp', () => {
  it('should match snapshot', () => {
    const mockProps = {
      dev: {name: 'tony'},
      history: {
        push: jest.fn()
      },
      signup: {
        status: false
      }
    };
    const wrapper = shallow(<SignUp {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('handleInputChange', () => {
    let mockProps;
    let mockEvent;
    let wrapper;

    beforeEach(() => {
      mockProps = {
        dev: {name: 'tony'},
        history: {
          push: jest.fn()
        },
        signup: {
          status: false
        }
      };
      mockEvent = {
        target: {
          name: "repo",
          value: 'input value'
        }
      };
      wrapper = shallow(<SignUp {...mockProps} />);
    });

    it('should setState off the params', () => {
      wrapper.instance().handleInputChange(mockEvent);
    
      expect(wrapper.state('repo')).toEqual(mockEvent.target.value);
    });
  });

  describe('handleInputChange', () => {
    let mockProps;
    let wrapper;

    beforeEach(() => {
      mockProps = {
        dev: {name: 'tony', id: 1},
        history: {
          push: jest.fn()
        },
        signup: {
          status: false,
          problemID: 123
        },
        handleSignup: jest.fn()
      };
      wrapper = shallow(<SignUp {...mockProps} />);
    });

    it('should call given functions', () => {
      wrapper.setState({
        repo: 'repo',
        contact: '@tony'
      });
      wrapper.instance().handleSubmit();
    
      expect(wrapper.instance().database.writeContributerToDatabase).toHaveBeenCalled();
    });
  });
});