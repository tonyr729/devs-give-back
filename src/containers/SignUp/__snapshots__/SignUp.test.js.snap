// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    dev={
        Object {
            "name": "tony",
          }
    }
    history={
        Object {
            "push": [Function],
          }
    }
    signup={
        Object {
            "status": false,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Redirect
          push={false}
          to="/dev-profile"
/>,
        <div
          className="signup-form"
>
          <p
                    className="signup-title"
          >
                    Awesome!
          </p>
          <p
                    className="signup-info"
          >
                    Please create a GitHub Repo for this project and prefered contact info for the client and other developers regarding this project.
          </p>
          <input
                    autoFocus={true}
                    className="repo-input"
                    name="repo"
                    onChange={[Function]}
                    placeholder="https://github.com/:user/:repo"
                    type="text"
                    value=""
          />
          <input
                    className="contact-info"
                    name="contact"
                    onChange={[Function]}
                    placeholder="Contact Info"
                    type="text"
                    value=""
          />
          <button
                    className="submit-signup"
                    onClick={[Function]}
          >
                    Submit!
          </button>
</div>,
      ],
      "className": "overlay",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "push": false,
          "to": "/dev-profile",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="signup-title"
>
              Awesome!
</p>,
            <p
              className="signup-info"
>
              Please create a GitHub Repo for this project and prefered contact info for the client and other developers regarding this project.
</p>,
            <input
              autoFocus={true}
              className="repo-input"
              name="repo"
              onChange={[Function]}
              placeholder="https://github.com/:user/:repo"
              type="text"
              value=""
/>,
            <input
              className="contact-info"
              name="contact"
              onChange={[Function]}
              placeholder="Contact Info"
              type="text"
              value=""
/>,
            <button
              className="submit-signup"
              onClick={[Function]}
>
              Submit!
</button>,
          ],
          "className": "signup-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Awesome!",
              "className": "signup-title",
            },
            "ref": null,
            "rendered": "Awesome!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Please create a GitHub Repo for this project and prefered contact info for the client and other developers regarding this project.",
              "className": "signup-info",
            },
            "ref": null,
            "rendered": "Please create a GitHub Repo for this project and prefered contact info for the client and other developers regarding this project.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": true,
              "className": "repo-input",
              "name": "repo",
              "onChange": [Function],
              "placeholder": "https://github.com/:user/:repo",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "contact-info",
              "name": "contact",
              "onChange": [Function],
              "placeholder": "Contact Info",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit!",
              "className": "submit-signup",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Submit!",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Redirect
            push={false}
            to="/dev-profile"
/>,
          <div
            className="signup-form"
>
            <p
                        className="signup-title"
            >
                        Awesome!
            </p>
            <p
                        className="signup-info"
            >
                        Please create a GitHub Repo for this project and prefered contact info for the client and other developers regarding this project.
            </p>
            <input
                        autoFocus={true}
                        className="repo-input"
                        name="repo"
                        onChange={[Function]}
                        placeholder="https://github.com/:user/:repo"
                        type="text"
                        value=""
            />
            <input
                        className="contact-info"
                        name="contact"
                        onChange={[Function]}
                        placeholder="Contact Info"
                        type="text"
                        value=""
            />
            <button
                        className="submit-signup"
                        onClick={[Function]}
            >
                        Submit!
            </button>
</div>,
        ],
        "className": "overlay",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/dev-profile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="signup-title"
>
                Awesome!
</p>,
              <p
                className="signup-info"
>
                Please create a GitHub Repo for this project and prefered contact info for the client and other developers regarding this project.
</p>,
              <input
                autoFocus={true}
                className="repo-input"
                name="repo"
                onChange={[Function]}
                placeholder="https://github.com/:user/:repo"
                type="text"
                value=""
/>,
              <input
                className="contact-info"
                name="contact"
                onChange={[Function]}
                placeholder="Contact Info"
                type="text"
                value=""
/>,
              <button
                className="submit-signup"
                onClick={[Function]}
>
                Submit!
</button>,
            ],
            "className": "signup-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Awesome!",
                "className": "signup-title",
              },
              "ref": null,
              "rendered": "Awesome!",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please create a GitHub Repo for this project and prefered contact info for the client and other developers regarding this project.",
                "className": "signup-info",
              },
              "ref": null,
              "rendered": "Please create a GitHub Repo for this project and prefered contact info for the client and other developers regarding this project.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoFocus": true,
                "className": "repo-input",
                "name": "repo",
                "onChange": [Function],
                "placeholder": "https://github.com/:user/:repo",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "contact-info",
                "name": "contact",
                "onChange": [Function],
                "placeholder": "Contact Info",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit!",
                "className": "submit-signup",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Submit!",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
